File: java/src/main/java/com/dokmatiq/docgen/model/DocumentRequest.java
package com.dokmatiq.docgen.model;

import com.fasterxml.jackson.annotation.JsonInclude;
import java.util.List;
import java.util.Map;

/** Main document generation request. */
@JsonInclude(JsonInclude.Include.NON_NULL)
public record DocumentRequest(
        String htmlContent,
        String markdownContent,
        String templateName,
        String templateBase64,
        Map<String, String> fields,
        Map<String, String> bookmarks,
        Map<String, String> markdownBookmarks,
        List<ImageData> images,
        List<QrCodeData> qrCodes,
        List<BarcodeData> barcodes,
        Map<String, TableData> tables,
        PageSettings pageSettings,
        Object watermark,
        StationeryConfig stationery,
        List<ContentArea> contentAreas,
        InvoiceData invoiceData,
        String password,
        OutputFormat outputFormat,
        String callbackUrl,
        String callbackSecret,
        Map<String, String> markdownStyles
) {}

File: dotnet/src/DocGen/Models/DocumentRequest.cs
using System.Text.Json.Serialization;

namespace Dokmatiq.DocGen.Models;

/// <summary>
/// Request for document generation.
/// </summary>
public class DocumentRequest
{
    [JsonPropertyName("htmlContent")]
    public string? HtmlContent { get; set; }

    [JsonPropertyName("markdownContent")]
    public string? MarkdownContent { get; set; }

    [JsonPropertyName("templateName")]
    public string? TemplateName { get; set; }

    [JsonPropertyName("templateBase64")]
    public string? TemplateBase64 { get; set; }

    [JsonPropertyName("fields")]
    public Dictionary<string, string>? Fields { get; set; }

    [JsonPropertyName("bookmarks")]
    public Dictionary<string, string>? Bookmarks { get; set; }

    [JsonPropertyName("markdownBookmarks")]
    public Dictionary<string, string>? MarkdownBookmarks { get; set; }

    [JsonPropertyName("images")]
    public List<ImageData>? Images { get; set; }

    [JsonPropertyName("qrCodes")]
    public List<QrCodeData>? QrCodes { get; set; }

    [JsonPropertyName("barcodes")]
    public List<BarcodeData>? Barcodes { get; set; }

    [JsonPropertyName("tables")]
    public Dictionary<string, TableData>? Tables { get; set; }

    [JsonPropertyName("pageSettings")]
    public PageSettings? PageSettings { get; set; }

    [JsonPropertyName("watermark")]
    public object? Watermark { get; set; }

    [JsonPropertyName("stationery")]
    public StationeryConfig? Stationery { get; set; }

    [JsonPropertyName("contentAreas")]
    public List<ContentArea>? ContentAreas { get; set; }

    [JsonPropertyName("invoiceData")]
    public InvoiceData? InvoiceData { get; set; }

    [JsonPropertyName("password")]
    public string? Password { get; set; }

    [JsonPropertyName("outputFormat")]
    public OutputFormat? OutputFormat { get; set; }

    [JsonPropertyName("callbackUrl")]
    public string? CallbackUrl { get; set; }

    [JsonPropertyName("callbackSecret")]
    public string? CallbackSecret { get; set; }

    [JsonPropertyName("markdownStyles")]
    public Dictionary<string, string>? MarkdownStyles { get; set; }
}
